azure_functions_worker/utils/dependency.py

```diff
@@ -0,0 +1,313 @@
+    )
+    def use_worker_dependencies(cls):
+        """Switch the sys.path and ensure the worker imports are loaded from
+        Worker's dependenciess.
+
+        This will not affect already imported namespaces, but will clear out
+        the module cache and ensure the upcoming modules are loaded from
+        worker's dependency path.
+        """
+
+        # The following log line will not show up in core tools but should
+        # work in kusto since core tools only collects gRPC logs. This function
+        # is executed even before the gRPC logging channel is ready.
+        cls._remove_from_sys_path(cls.cx_deps_path)
+        cls._remove_from_sys_path(cls.cx_working_dir)
+        cls._add_to_sys_path(cls.worker_deps_path, True)
+        logger.info(f'Start using worker dependencies {cls.worker_deps_path}')
+
+    @classmethod
+    @enable_feature_by(
+        flag=PYTHON_ISOLATE_WORKER_DEPENDENCIES,
+        flag_default=PYTHON_ISOLATE_WORKER_DEPENDENCIES_DEFAULT_39
+    )
+    def use_customer_dependencies(cls):
+        """Switch the sys.path and ensure the customer's code import are loaded
+        from CX's deppendencies.
+
+        This will not affect already imported namespaces, but will clear out
+        the module cache and ensure the upcoming modules are loaded from
+        customer's dependency path.
+
+        As for Linux Consumption, this will only remove worker_deps_path,
+        but the customer's path will be loaded in function_environment_reload.
+
+        The search order of a module name in customer frame is:
+        1. cx_deps_path
+        2. cx_working_dir
+        """
+        cls._remove_from_sys_path(cls.worker_deps_path)
+        cls._add_to_sys_path(cls.cx_deps_path, True)
+        cls._add_to_sys_path(cls.cx_working_dir, False)
+        logger.info(f'Start using customer dependencies {cls.cx_deps_path}')
+
+    @classmethod
+    def reload_azure_google_namespace(cls, cx_working_dir: str):
+        """Reload azure and google namespace, this including any modules in
+        this namespace, such as azure-functions, grpcio, grpcio-tools etc.
+
+        Depends on the PYTHON_ISOLATE_WORKER_DEPENDENCIES, the actual behavior
+        differs.
+
+        Parameters
+        ----------
+        cx_working_dir: str
+            The path which contains customer's project file (e.g. wwwroot).
+        """
+        use_new_env = os.getenv(PYTHON_ISOLATE_WORKER_DEPENDENCIES)
+        if use_new_env is None:
+            use_new = PYTHON_ISOLATE_WORKER_DEPENDENCIES_DEFAULT_39
+        else:
+            use_new = is_true_like(use_new_env)
+
+        if use_new:
+            cls.reload_azure_google_namespace_from_customer_deps(
+                cx_working_dir
+            )
+        else:
+            cls.reload_azure_google_namespace_from_worker_deps()
+
+    @classmethod
+    def reload_azure_google_namespace_from_worker_deps(cls):
+        """This is the old implementation of reloading azure and google
```

Comment:
I think we should still keep the old implementation and it is specific to linux consumption. Since there're already a lot of function apps using google.protobuf and azure.functions which are bumped to a newer version that our worker is using.

For instance, if we roll it out by default, it potentially breaks apps with `protobuf=<old version>` in their requirements.txt, since now the app will load protobuf from .python_packages instead of worker deps.